src/main/java/mate/academy/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package mate.academy.service;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.CartItemQuantityRequestDto;
+import mate.academy.dto.CartItemRequestDto;
+import mate.academy.dto.ShoppingCartDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.CartItemMapper;
+import mate.academy.mapper.ShoppingCartMapper;
+import mate.academy.model.Book;
+import mate.academy.model.CartItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.repository.CartItemRepository;
+import mate.academy.repository.ShoppingCartRepository;
+import mate.academy.repository.UserRepository;
+import mate.academy.repository.book.BookRepository;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    @Transactional
+    public void addItemToCart(User user,
+                              CartItemRequestDto cartItemRequestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseGet(() -> registerNewShoppingCart(user));
+
+        Optional<CartItem> existingCartItem = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().getId().equals(cartItemRequestDto.getBookId()))
+                .findFirst();
+        existingCartItem.ifPresentOrElse(
+                cartItem -> {
+                    cartItem.setQuantity(cartItemRequestDto.getQuantity() + cartItem.getQuantity());
+                    cartItemRepository.save(cartItem);
+                },
+                () -> {
+                    CartItem newCartItem = createNewCartItem(cartItemRequestDto, shoppingCart);
+                    cartItemRepository.save(newCartItem);
+                }
+        );
```

Comment:
As for me in this case classical structure is more simple and readable than functional style one
```suggestion
        CartItem cartItem;
        if (existingCartItem.isEmpty) {
            cartItem = createNewCartItem(cartItemRequestDto, shoppingCart);
        } else {
           cartItem = existingCartItem.get();
           cartItem.setQuantity(cartItemRequestDto.getQuantity() + cartItem.getQuantity());
        }
        cartItemRepository.save(cartItem);
```